### ISSUE TITLE
cancel a compression in progress [feature request]

### ISSUE THREAD

**Issue by felipeaf** [Donaldcwl/browser-image-compression#101]
Some compressions (very large resolutions and bytes in size) can take a while and be CPU expensive. It should be a way to cancel a compression in progress in WebWorker mode (example of use cases: in SPA, when user remove file from form or change file again).

**Comment by handbremse** [comment]
Something like a cancelToken that contains a Promise, which can be resolved/rejected by the user.

**Comment by Donaldcwl** [comment]
In the coming releases, we will use AbortController for canceling mechanism:
https://developer.mozilla.org/en-US/docs/Web/API/AbortController

reference: https://axios-http.com/docs/cancellation

**Comment by Donaldcwl** [comment]
This feature is implemented in v2.0.0. For example:

```javascript
function handleImageUpload(event) {

  var imageFile = event.target.files[0];

  var controller = new AbortController();

  var options = {
    // other options here
    signal: controller.signal,
  }
  imageCompression(imageFile, options)
    .then(function (compressedFile) {
      return uploadToServer(compressedFile); // write your own logic
    })
    .catch(function (error) {
      console.log(error.message); // output: I just want to stop
    });
  
  // simulate abort the compression after 1.5 seconds
  setTimeout(function () {
    controller.abort(new Error('I just want to stop'));
  }, 1500);
}
```
